src/firecracker/src/api_server_adapter.rs

```diff
@@ -151,36 +165,25 @@ pub(crate) fn run_with_api(
     let api_thread = thread::Builder::new()
         .name("fc_api".to_owned())
         .spawn(move || {
-            let res = ApiServer::new(to_vmm, from_vmm, to_vmm_event_fd).bind_and_run(
+            match ApiServer::new(to_vmm, from_vmm, to_vmm_event_fd).bind_and_run(
                 &api_bind_path,
                 process_time_reporter,
                 &api_seccomp_filter,
                 api_payload_limit,
                 socket_ready_sender,
-            );
-
-            let Err(err) = res else {
-                return;
-            };
```

Comment:
If we change this to 
```rust
            let Err(err) = res else {
                return Ok(());
            };
```
we can keep the existing pattern (which I implemented and personally prefer although this is opinion).

Reply:
I wonder why you prefer this way over usual match. (except part that you implemented it)
But ultimately yeah, I can change it back.

Reply:
It simplifies the match (I don't really like `Ok(_) => Ok(())`).